cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -158,11 +158,25 @@ function create_resource_from_string() {
   local -r config_name=$4;
   local -r namespace=$5;
   while [ "${tries}" -gt 0 ]; do
+    # Resources that are set to the addon mode EnsureExists should not be overwritten if they already exist.
+    # EnsureExists means we should use `kubectl create`, other modes mean we should use `kubectl apply`.
+    local verb="apply"
+    if echo "${config_string}" | grep --silent "^\s*${ADDON_MANAGER_LABEL}:\s\+EnsureExists"; then
```

Comment:
/hold 

does this work if the string contains multiple objects, some with EnsureExists and some without?

why is a different approach being taken here than in ensure_addons/reconcile_addons (which use label selectors)

Reply:
It wouldn't work, good point. There don't seem to be any multi-object files in testing.

I seemed to be under the impression that the label selectors wouldn't work as intended previously, and I don't have any notes on why I thought that was the case. I will note that this image is currently bundled with kubectl version 1.13.2.

I'll test a change with separate `kubectl apply`/`kubectl create` commands with label selectors.